flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBase.java

```diff
@@ -40,26 +42,42 @@
 	protected transient Cluster cluster;
 	protected transient Session session;
 
-	protected transient Throwable exception = null;
+	protected transient AtomicReference<Throwable> exception;
```

Comment:
Does `exception` have to be an `AtomicReference`, or does a volatile variable suffice?